### PR TITLE
Run openshift aggregated behind kube

### PR DESCRIPTION
**Pull request by deads2k**
This is ready for review.  I've split some bits out into separate pulls to make it easier to review.  It separates openshift-apiserver and kube-apiserver and successfully produces a working cluster by aggregating openshift-apiserver using cluster up (on my machine).

### CODE REVIEW

**Comment by deads2k** [comment]
ah, I need hypershift in an image before service-catalog will pass.

**Comment by deads2k** [comment]
@tnozicka time to learn troubleshooting in the brave new world :)  Can you see why https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/18652/test_pull_request_origin_end_to_end/10175/ is failing?  The openshift controllers are now in a container and the logs appear to be available.  Looks to me like it was doing things though.

**Comment by deads2k** [comment]
@mfojtik squashed into reasonable commits, separate pulls split out if you want them https://github.com/openshift/origin/pull/18836 and https://github.com/openshift/origin/pull/18838 .

I think this is a reasonable spot to start.

**Comment by deads2k** [comment]
docker problem

```
   Error: failed to install "openshift-infra/template service broker registration": Error response from daemon: devmapper: Error activating devmapper device for '39510e88daa2e662c3ef558814d8216ffe8ce3135d0923ed69dda411add92b01': devicemapper: Error running deviceCreate (ActivateDevice) dm_task_run failed
```
 https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/18652/test_pull_request_origin_extended_clusterup/508/

/retest

**Comment by deads2k** [comment]
/retest

service catalog won't pass until we get the hypershift commands.

File: install/openshift-apiserver/install.yaml [hunk 1]
```diff
@@ -0,0 +1,266 @@
+# TODO update this to use a template and then install via a container running the process step locally
```
Comment (mfojtik) [inline install/openshift-apiserver/install.yaml]: can you create follow up for this?

**Comment by deads2k** [comment]
service-catalog tests need `UPSTREAM: <carry>: aggregator to proxy oapi to apps.openshift.io server` separately to pass because they don't use the images built as part of this pull.

@jpeeler @jboyd01 I can't recall which of you owns this test.  Why aren't we use using the level of openshift from the pull for the images?

**Comment by jpeeler** [comment]
@deads2k I have a PR here to modify the daily job https://github.com/openshift/aos-cd-jobs/pull/1191. I'm testing it now and if it works ok, I'll adjust the pull request test to do the same.

**Comment by deads2k** [comment]
> @deads2k I have a PR here to modify the daily job openshift/aos-cd-jobs#1191. I'm testing it now and if it works ok, I'll adjust the pull request test to do the same.

Thanks

**Comment by deads2k** [comment]
@mfojtik comment addressed

**Comment by deads2k** [comment]
/hold

got a wrong commit in here somehow

**Comment by deads2k** [comment]
/hold cancel

fixed

File: pkg/oc/bootstrap/clusterup/staticpods/load.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 	"github.com/openshift/origin/pkg/oc/bootstrap"
 )
 
-func substitute(in string, replacements map[string]string) string {
+func Substitute(in string, replacements map[string]string) string {
```
Comment (mfojtik) [inline pkg/oc/bootstrap/clusterup/staticpods/load.go]: why making this public?

File: vendor/k8s.io/kubernetes/staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go [hunk 3]
```diff
@@ -284,6 +284,12 @@ func (s *APIAggregator) AddAPIService(apiService *apiregistration.APIService) er
 	s.GenericAPIServer.Handler.NonGoRestfulMux.Handle(proxyPath, proxyHandler)
 	s.GenericAPIServer.Handler.NonGoRestfulMux.UnlistedHandlePrefix(proxyPath+"/", proxyHandler)
 
+	enableOapiHack := true
```
Comment (soltysh) [inline vendor/k8s.io/kubernetes/staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go]: What's the point of this?

Comment (mfojtik) [reply vendor/k8s.io/kubernetes/staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go]: was asking @deads2k the same... this is to proxy securely to /oapi as there is no way to register that in the apiservice API.... so this is fine and we should remove this when we deprecate /oapi...

**Comment by deads2k** [comment]
comments addressed.

**Comment by jpeeler** [comment]
/test service-catalog
Job is building the all origin images now.

**Comment by deads2k** [comment]
> Job is building all the origin images now.

@jpeeler Excellent! Thanks.

File: vendor/k8s.io/kubernetes/staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go (hunk 3 above)
Comment (soltysh) [reply vendor/k8s.io/kubernetes/staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go]: Yeah, I get that, what I was asking why we need this true variable which is checked right in the next line?

**Comment by soltysh** [comment]
I have only one question, but overall this lgtm.

**Comment by deads2k** [comment]
/hold

SC is still having trouble.  passes in "normal" run though.

**Comment by deads2k** [comment]
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/18652/test_pull_request_origin_service_catalog/194/ appears to be failing because the origin image does not include `UPSTREAM: <carry>: aggregator to proxy oapi to apps.openshift.io server` from this pull.  I can reproduce the problem if I run with current --version=latest.

It fails on an invocation like `oc cluster up --version=latest --service-catalog`.  We can see `oc cluster up --service-catalog --loglevel=5 --server-loglevel=5 --version=7eb90cc` work in this test here https://storage.googleapis.com/origin-ci-test/pr-logs/pull/18652/test_pull_request_origin_extended_clusterup/530/build-log.txt when run with the images produced by this pull.

**Comment by jpeeler** [comment]
Merging this from a service catalog standpoint is fine. The image usage is not properly sorted out yet.

**Pull request merged**